salt/state.py

```diff
@@ -2813,29 +2855,15 @@ def render_state(self, sls, saltenv, mods, matches, local=False):
                         continue
 
                     if inc_sls.startswith('.'):
-                        match = re.match(r'^(\.+)(.*)$', inc_sls)
-                        if match:
-                            levels, include = match.groups()
-                        else:
-                            msg = ('Badly formatted include {0} found in include '
-                                    'in SLS \'{2}:{3}\''
-                                    .format(inc_sls, saltenv, sls))
-                            log.error(msg)
-                            errors.append(msg)
-                            continue
-                        level_count = len(levels)
-                        p_comps = sls.split('.')
-                        if state_data.get('source', '').endswith('/init.sls'):
-                            p_comps.append('init')
-                        if level_count > len(p_comps):
-                            msg = ('Attempted relative include of \'{0}\' '
-                                   'within SLS \'{1}:{2}\' '
-                                   'goes beyond top level package '
-                                   .format(inc_sls, saltenv, sls))
-                            log.error(msg)
-                            errors.append(msg)
+                        try:
+                            source_file = state_data.get('source', '')
+                            _sls = sls
+                            if source_file.endswith('/init.sls'):
+                                _sls += '.init'
```

Comment:
Unfortunately, since `foo/init.sls`' `sls` is just `foo`, relative `include`s like `.bar` would end up matching `bar`, not `foo.bar`, which is why we need this `+= '.init'`, but this is only available if we have `state_data` (which is also only true if we're not running locally, according to [line 2776](https://github.com/saltstack/salt/pull/32469/files#diff-e5cf22153489a6c40864babf3ca18002R2776)... bug?).

Is there a way to get the name of the file a state came from in `compile_high_data`? Despite poking around for a couple hours today, I wasn't able to find any definitive answers. [`render_template`](https://github.com/saltstack/salt/blob/develop/salt/state.py#L2287-L2291) is the closest thing I could find, but there's no indication as to why those lines were commented out (they were initially added commented out).